spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/RabbitListenerContainerFactory.java

```diff
@@ -48,4 +49,17 @@ default C createListenerContainer() {
 		return createListenerContainer(null);
 	}
 
+	/**
+	 * set a bean name.
+	 * @since 3.2
+	 */
+	void setBeanName(String name);
+
+	/**
+	 * Return a bean name of the component or null if not a bean.
+	 * @return the bean name.
+	 * @since 3.2
+	 */
+	String getBeanName();
```

Comment:
No. They have to be `default` to avoid a breaking change in this minor version.
This way the whole interface would still be a `@FunctionalInterface`.

Plus mark this method as `@Nullable`.